File: agent/api/v1/payload/references.py
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel


class IndexStatusResponse(StrEnum):
    pending = "pending"
    processing = "processing"
    completed = "completed"
    failed = "failed"


class ReferenceResponse(BaseModel):
    id: str
    collection_id: str
    filename: str
    doc_name: str
    content_type: str
    status: IndexStatusResponse
    error_message: str | None = None
    metadata: dict[str, object] | None = None
    created_at: datetime
    updated_at: datetime


class ReferenceChunkItem(BaseModel):
    id: str
    text: str
    metadata: dict[str, object]


class ReferenceChunksResponse(BaseModel):
    items: list[ReferenceChunkItem]

File: frontend/src/types/references.ts
import type { CollectionId } from './conversations';

export type ReferenceId = string;

export const REFERENCE_STATUS = {
    PENDING: 'pending',
    PROCESSING: 'processing',
    COMPLETED: 'completed',
    FAILED: 'failed',
} as const;

export type ReferenceStatus = (typeof REFERENCE_STATUS)[keyof typeof REFERENCE_STATUS];

export interface Reference {
    id: ReferenceId;
    collectionId: CollectionId;
    docName: string;
    filename: string;
    contentType: string;
    status: ReferenceStatus;
    errorMessage: string | null;
    metadata: Record<string, unknown> | null;
    createdAt: string;
    updatedAt: string;
}

/** Display name for UI rows */
export function referenceDisplayName(ref: Reference): string {
    return ref.docName || ref.filename;
}
